ILEditing/MechanicILChanges.cs

```diff
@@ -765,6 +767,27 @@ private static void DrawForegroundParticles(Terraria.On_Main.orig_DrawInfernoRin
         #endregion
 
         #region Custom Lava Visuals
+
+        private static void CacheLavaStyle(ILContext il)
+        {
+            ILCursor cursor = new ILCursor(il);
+            cursor.EmitDelegate(() =>
+            {
+                // Immediately cache the lava drawing style.
+                // This will pay off in SPADES when we go to draw the tiles.
+                foreach (CustomLavaStyle style in CustomLavaManagement.CustomLavaStyles)
+                {
+                    if (style.ChooseLavaStyle())
+                    {
+                        cachedLavaStyle = style;
+                        return;
+                    }
+                }
+
+                cachedLavaStyle = default;
+            });
+        }
```

Comment:
We could probably prefer a detour over an IL edit here.

Reply:
Edits made, thanks for the suggestion!